src/3d/qgspointcloudlayerchunkloader_p.cpp

```diff
@@ -255,4 +256,107 @@ QgsPointCloudLayerChunkedEntity::~QgsPointCloudLayerChunkedEntity()
   cancelActiveJobs();
 }
 
+QVector<RayHit> QgsPointCloudLayerChunkedEntity::rayIntersection( const QgsRay3D &ray, const RayCastContext &context ) const
+{
+
+  QgsPointCloudLayerChunkLoaderFactory *factory = static_cast<QgsPointCloudLayerChunkLoaderFactory *>( mChunkLoaderFactory );
+
+  QVector<RayHit> result;
+  // transform ray
+  const QgsVector3D originMapCoords = factory->mMap.worldToMapCoordinates( ray.origin() );
+  const QgsVector3D pointMapCoords = factory->mMap.worldToMapCoordinates( ray.origin() + ray.origin().length() * ray.direction().normalized() );
+  QgsVector3D directionMapCoords = pointMapCoords - originMapCoords;
+  directionMapCoords.normalize();
+
+  const QVector3D rayOriginMapCoords( originMapCoords.x(), originMapCoords.y(), originMapCoords.z() );
+  const QVector3D rayDirectionMapCoords( directionMapCoords.x(), directionMapCoords.y(), directionMapCoords.z() );
```

Comment:
this looks dangerous - we're switching to floats here, but using "big" coordinates - we will probably loose precision. I think we should have a rule that map coordinates are _always_ in doubles, only world coordinates are allowed as floats (as they are centered around zero).

Reply:
This was the existing implementation in Qgs3DMapToolIdentify. I like the rule, I'll modify this to follow it.